lib/typeprof/core/ast/control.rb

```diff
@@ -392,11 +392,75 @@ def undefine0(genv)
 
       def install0(genv)
         ret = Vertex.new(self)
+
+        vars = []
+        @body.modified_vars(@lenv.locals.keys, vars) if @body
+        @rescue_clauses.each do |clause|
+          clause.modified_vars(@lenv.locals.keys, vars)
+        end
+        @else_clause.modified_vars(@lenv.locals.keys, vars) if @else_clause
+        @ensure_clause.modified_vars(@lenv.locals.keys, vars) if @ensure_clause
+        vars.uniq!
+
+        old_vtxs = {}
+        vars.each do |var|
+          vtx = @lenv.get_var(var)
+          old_vtxs[var] = vtx
+        end
+
         @changes.add_edge(genv, @body.install(genv), ret)
-        @rescue_conds.each {|cond| cond.install(genv) }
-        @rescue_clauses.each {|clause| @changes.add_edge(genv, clause.install(genv), ret) }
-        @changes.add_edge(genv, @else_clause.install(genv), ret) if @else_clause
+
+        body_vtxs = {}
+        vars.each do |var|
+          body_vtxs[var] = @lenv.get_var(var)
+        end
+
+        @rescue_conds.each { |cond| cond.install(genv) }
+
+        rescue_vtxs = []
+        @rescue_clauses.each do |clause|
+          vars.each do |var|
+            old_vtx = old_vtxs[var]
+            nvtx = old_vtx.new_vertex(genv, self)
+
+            @changes.add_edge(genv, body_vtxs[var], nvtx) if body_vtxs[var]
```

Comment:
```suggestion
            @changes.add_edge(genv, body_vtxs[var], nvtx) unless body_vtxs[var] == old_vtxs[var]
```

I couldn't figure out the case where `body_vtxs[var]` would be falsy.
Also, I think this guard is not needed if lines 398..402 were deleted.

Reply:
398...402 were deleted and are no longer needed.